utils/reconciler.go

```diff
@@ -265,17 +311,42 @@ func (r *ReconcilerBase) ManageError(issue error, conditionType common.StatusCon
 // ManageSuccess ...
 func (r *ReconcilerBase) ManageSuccess(conditionType common.StatusConditionType, ba common.BaseComponent) (reconcile.Result, error) {
 	s := ba.GetStatus()
+	oldRecCondition := s.GetCondition(conditionType)
 
-	statusCondition := s.NewCondition(conditionType)
-	statusCondition.SetReason("")
-	statusCondition.SetMessage("")
-	statusCondition.SetStatus(corev1.ConditionTrue)
-	s.SetCondition(statusCondition)
+	newRecCondition := s.NewCondition(conditionType)
+	newRecCondition.SetReason("")
+	newRecCondition.SetMessage("")
+	newRecCondition.SetStatus(corev1.ConditionTrue)
+	s.SetCondition(newRecCondition)
 
 	addStatusWarnings(ba)
 
 	//Check application status (reconciliation & resource status & endpoint status)
-	readyStatus := r.CheckApplicationStatus(ba)
+	oldAppCondition, newAppCondition := r.CheckApplicationStatus(ba)
+
+	var retryInterval time.Duration
+
+	// If resources are not ready
+	if newAppCondition.GetStatus() != corev1.ConditionTrue {
+		// If the resources were ready and now they are not or encountered a different error
+		if oldAppCondition == nil || oldAppCondition.GetStatus() != newAppCondition.GetStatus() || oldAppCondition.GetMessage() != newAppCondition.GetMessage() {
+			retryInterval = getBaseReconcileInterval(newAppCondition, s)
+		} else {
+			// If the resources stay unready and the error message has not changed
+			maxSeconds := 240 // Max 240 minutes
+			retryInterval = updateReconcileInterval(maxSeconds, oldAppCondition, newAppCondition, s)
+		}
+	} else {
+		// If the application and resources are ready
+		// If the application or the resources were not ready before
+		if oldRecCondition == nil || oldRecCondition.GetStatus() != newRecCondition.GetStatus() || oldRecCondition.GetMessage() != newRecCondition.GetMessage() {
```

Comment:
Just wanted to make sure checking the reconciled condition, instead of app condition, is intentional here